validator/accounts/v2/new.go

```diff
@@ -0,0 +1,265 @@
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var log = logrus.WithField("prefix", "accounts-v2")
+
+const minPasswordLength = 8
+
+var keymanagerKindSelections = map[v2keymanager.Kind]string{
+	v2keymanager.Direct:  "Direct, On-Disk Accounts (Recommended)",
+	v2keymanager.Derived: "Derived Accounts (Advanced)",
+	v2keymanager.Remote:  "Remote Accounts (Advanced)",
+}
+
+// New creates a new validator account from user input. If a user
+// does not have an initialized wallet at the specified wallet path, this
+// method will create a new wallet and ask user for input for their new wallet's
+// available options.
+func New(cliCtx *cli.Context) error {
+	// Read a wallet's directory from user input.
+	walletDir := inputWalletDir(cliCtx)
+
+	// Read the directory for password storage from user input.
+	passwordsDirPath := inputPasswordsDirectory(cliCtx)
+
+	ctx := context.Background()
+	// Check if the user has a wallet at the specified path.
+	// If a user does not have a wallet, we instantiate one
+	// based on specified options.
+	var wallet *Wallet
+	var err error
+	var isNewWallet bool
+	ok, err := hasWalletDir(walletDir)
+	if err != nil {
+		log.Fatalf("Could not check if wallet exists at %s: %v", walletDir, err)
+	}
+	if ok {
+		// Read the wallet from the specified path.
+		wallet, err = ReadWallet(ctx, &WalletConfig{
+			PasswordsDir: passwordsDirPath,
+			WalletDir:    walletDir,
+		})
+		if err != nil {
+			log.Fatalf("Could not read wallet at specified path %s: %v", walletDir, err)
+		}
+	} else {
+		// Determine the desired keymanager kind for the wallet from user input.
+		keymanagerKind := inputKeymanagerKind(cliCtx)
+
+		walletConfig := &WalletConfig{
+			PasswordsDir:   passwordsDirPath,
+			WalletDir:      walletDir,
+			KeymanagerKind: keymanagerKind,
+		}
+		wallet, err = CreateWallet(ctx, walletConfig)
+		if err != nil {
+			log.Fatalf("Could not create wallet at specified path %s: %v", walletDir, err)
+		}
+		isNewWallet = true
+	}
+
+	// We initialize a new keymanager depending on the user's selected keymanager kind.
+	keymanager := initializeWalletKeymanager(ctx, wallet, isNewWallet)
+
+	// Read the new account's password from user input.
+	password := inputAccountPassword(cliCtx)
+
+	// Create a new validator account using the specified keymanager.
+	// TODO(#6220): Implement.
+	if err := keymanager.CreateAccount(ctx, password); err != nil {
+		log.Fatalf("Could not create account in wallet: %v", err)
+	}
+	return nil
+}
+
+// Initializes a keymanager. If a config file exists in the wallet, it
+// reads the config file and initializes the keymanager that way. Otherwise,
+// writes a new configuration file to the wallet and returns the initialized
+// keymanager for use.
+func initializeWalletKeymanager(ctx context.Context, wallet *Wallet, isNewWallet bool) v2keymanager.IKeymanager {
+	var keymanager v2keymanager.IKeymanager
+	var err error
+	if isNewWallet {
```

Comment:
instead of using a boolean here, can we just spin it off into its own separate function for new wallets ?